h/static/scripts/controllers/confirm-submit-controller.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+var Controller = require('../base/controller');
+
+/* Turn a normal submit element into one that shows a confirm dialog.
+ *
+ * The element's form will only be submitted if the user answers the
+ * confirmation dialog positively.
+ *
+ */
+class ConfirmSubmitController extends Controller {
+  constructor(element, options) {
+    super(element);
+
+    var window_ = options.window || window;
```

Comment:
A couple of people have asked me about usage of const/let/var recently. I suggest that in new source files we follow the Airbnb style - `const` where possible, `let` otherwise. There are tools to automate updating existing code and I'll look into using those so we're consistent everywhere.